### ISSUE TITLE
Code breaks on receiving "user_change" event

### ISSUE THREAD

**Issue by gaurav2493** [slackapi/java-slack-sdk#406]
### Issue Type

- [x] Bug
- [ ] Enhancement / Feature request
- [ ] Question
- [ ] Documentation

### Description

When user_change event is sent, the code breaks with the following exception

```
java.lang.UnsupportedOperationException: JsonObject

	at com.google.gson.JsonElement.getAsString(JsonElement.java:179)
	at com.slack.api.bolt.middleware.builtin.IgnoringSelfEvents.apply(IgnoringSelfEvents.java:79)
```

This happens because code expects `user` to contain string but it contained jsonObject

```
{
    "token": ".....",
    "team_id": ".....",
    "enterprise_id": "....",
    "api_app_id": "....",
    "event": {
        "type": "user_change",
        "user": {
            "id": "....",
            "name": "gaurav",
......
}
```

---

The issue is reproducible in:
* Module Version: 1.0.0
* JDK Version: 1.8.0_231

The steps to reproduce are:
1. subscribe to user_change event
2. Change any field to produce webhook

The expected result is:
Event to get handled

The actual result is:
exception is thrown

---

* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/java-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

**Comment by seratch** [comment]
@gaurav2493 Hello, thank you for flagging this (and fixing it 🙇). Your fix looks perfect! This one is so critical that I will release a new version (version 1.0.1) after merging your PR today.

**Comment by seratch** [comment]
@gaurav2493 Thanks again for your contribution here. I just published version 1.0.1 jars to the Sonatype OSS repositories. They'll be synchronized to Maven Central soon.
